debugger/debugger.c

```diff
@@ -62,67 +65,67 @@ bool exec_cmd (Emulator *emu, char *line, int len)
       }
 
       for (i = 0;i < steps;i++) {
-	decode(emu, fetch(emu), EXECUTE);
+	      decode(emu, fetch(emu, true), EXECUTE);
 
-        // Handle Peripherals     
-	handle_bcm(emu);
-	handle_timer_a(emu);
-	handle_port_1(emu);
-	handle_usci(emu); 
+        // Handle Peripherals
+	      handle_bcm(emu);
+	      handle_timer_a(emu);
+	      handle_port_1(emu);
+	      handle_usci(emu);
       }
 
       display_registers(emu);
       disassemble(emu, cpu->pc, 1);
-  }                                
-  
+  }
+
   // Quit program //
   else if ( !strncasecmp("quit", cmd, sizeof "quit") ||
-	    !strncasecmp("q", cmd, sizeof "q")) 
-    {    
+	    !strncasecmp("q", cmd, sizeof "q"))
+    {
       // This flag stops the main loop in main.c
       deb->quit = true;
     }
-  
+
   // run the program until a breakpoint is hit //
   else if ( !strncasecmp("run", cmd, sizeof "run") ||
-	    !strncasecmp("r", cmd, sizeof "r")) 
+	    !strncasecmp("r", cmd, sizeof "r"))
     {
       cpu->running = true;
       deb->debug_mode = false;
 
-      update_register_display(emu);
+      //update_register_display(emu);
     }
 
   // Display disassembly of N at HEX_ADDR: dis [N] [HEX_ADDR] //
   else if ( !strncasecmp("disas", cmd, sizeof "disas") ||
 	    !strncasecmp("dis", cmd, sizeof "dis") ||
-	    !strncasecmp("disassemble", cmd, sizeof "disassemble")) 
+	    !strncasecmp("disassemble", cmd, sizeof "disassemble"))
     {
       uint16_t start_addr = cpu->pc;
-      uint32_t num = 10; 
-      
+      uint32_t num = 10;
+
       ops = sscanf(line, "%s %u %X", bogus1, &bogus2, &bogus3);
-      
+
       if (ops == 2) {
 	sscanf(line, "%s %u", bogus1, &num);
       }
       else if (ops == 3) {
-	sscanf(line, "%s %u %X", bogus1, &num, 
+	sscanf(line, "%s %u %X", bogus1, &num,
 		     (unsigned int *)&start_addr);
```

Comment:
This cast is bad. `start_addr` is uint16_t. Possible memory corruption here.

Reply:
Ack